src/fdb5/toc/TocStore.cc

```diff
@@ -241,11 +252,33 @@ void TocStore::moveTo(const Key& key, const Config& config, const eckit::URI& de
             }
             closedir(dirp);
 
-            eckit::ThreadPool pool("store"+dest_db.asString(), threads);
-            for (auto it = files.begin(); it != files.end(); it++) {
-                    pool.push(it->second);
+            if (mpi) { // dispath tasks with MPI
+                eckit::mpi::Comm& comm = eckit::mpi::comm();
+                ASSERT(comm.rank() == 0);
+
+                int ready;
+                char task[1024];
+
+                for (auto it = files.begin(); it != files.end(); it++) {
+                    eckit::mpi::Status status = comm.template receive<int>(&ready, 1, -1, -1);
+
+                    it->second->str(task, 1024);
```

Comment:
This function ought to be something like encode().

If it were me, I would use the eckit::serialisation stuff to encode into a buffer.